src/math.hpp

```diff
@@ -5,79 +5,65 @@
 
 #include <cmath>
 #include <cstddef>
+#include <iostream>
 #include <type_traits>
 
 namespace opt {
 
-template <Arithmetic T>  // clang-format off
-  requires(not Dual<T>)
-constexpr auto exp(T x)  // clang-format on
+template <class T>  // clang-format off
+  requires(Arithmetic<T> || Dual<T>)
+constexpr auto exp(T const& x)  // clang-format on
 {
     auto acc = decltype(x){1};
     auto power = acc;
     auto factorial = acc;
 
-    constexpr std::size_t taylor_terms{10U};
+    constexpr std::size_t taylor_terms{15U};
     for (std::size_t i{1U}; i < taylor_terms; ++i) {
-        power *= x;
-        factorial *= static_cast<decltype(factorial)>(i);
+        power = power * x;
+        factorial = factorial * T{static_cast<float>(i)};
```

Comment:
Should we implement implicit construction of a `dual<Arithmetic>` from `int`? The conversion to `float` may not work for a user-defined type.

Reply:
Right. I can do that tomorrow